homeassistant/const.py

```diff
@@ -752,11 +752,11 @@ class Platform(StrEnum):
 ENTITY_CATEGORY_CONFIG: Final = "config"
 ENTITY_CATEGORY_DIAGNOSTIC: Final = "diagnostic"
 ENTITY_CATEGORY_SYSTEM: Final = "system"
-ENTITY_CATEGORIES: Final[list[str]] = [
+ENTITY_CATEGORIES: Final[set[str]] = {
```

Comment:
Do you realize these are deprecated?

Reply:
Nope.  

I'm working on adding support in HomeKit and following what was done in alexa and google 

```
homeassistant/components/alexa/smart_home_http.py:from homeassistant.const import CONF_CLIENT_ID, CONF_CLIENT_SECRET, ENTITY_CATEGORIES
homeassistant/components/alexa/smart_home_http.py:            auxiliary_entity = registry_entry.entity_category in ENTITY_CATEGORIES
homeassistant/components/cloud/alexa_config.py:from homeassistant.const import CLOUD_NEVER_EXPOSED_ENTITIES, ENTITY_CATEGORIES
homeassistant/components/cloud/alexa_config.py:            auxiliary_entity = registry_entry.entity_category in ENTITY_CATEGORIES
homeassistant/components/cloud/google_config.py:from homeassistant.const import CLOUD_NEVER_EXPOSED_ENTITIES, ENTITY_CATEGORIES
homeassistant/components/cloud/google_config.py:            auxiliary_entity = registry_entry.entity_category in ENTITY_CATEGORIES
homeassistant/components/esphome/__init__.py:ENTITY_CATEGORIES: EsphomeEnumMapper[
homeassistant/components/esphome/__init__.py:        return ENTITY_CATEGORIES.from_esphome(self._static_info.entity_category)
homeassistant/components/google_assistant/http.py:from homeassistant.const import CLOUD_NEVER_EXPOSED_ENTITIES, ENTITY_CATEGORIES
homeassistant/components/google_assistant/http.py:            auxiliary_entity = registry_entry.entity_category in ENTITY_CATEGORIES

```

Reply:
`EntityCategory` is an enum now that can be used instead

Reply:
I guess I could move the constant there since we don't want to import `homeassistant.helpers.entity` into `homeassistant.const`